sdk/core/Azure.Core/src/ResponseClassifier.cs

```diff
@@ -47,18 +43,101 @@ public virtual bool IsRetriableException(Exception exception)
         /// </summary>
         public virtual bool IsRetriable(HttpMessage message, Exception exception)
         {
+            if (exception is null)
+            {
+                return IsErrorResponse(message);
+            }
+
+            // In order to allow users to override IsRetriableException logic,
+            // we must call through to that rather than using the base type
+            // default implementation.
+
             return IsRetriableException(exception) ||
-                   // Retry non-user initiated cancellations
-                   (exception is OperationCanceledException && !message.CancellationToken.IsCancellationRequested);
+                // Retry non-user initiated cancellations
+                (exception is OperationCanceledException &&
+                !message.CancellationToken.IsCancellationRequested);
```

Comment:
I'm not sure if this distinction is important, but should `IsRetriable` be concerned with cancellation, considering that most other places in the pipeline, including the retry policy, will already handle that case?

If it should, does it make more sense to check that first before everything as it supersedes any of the other checks above?

Reply:
@christothes, if I'm understanding your suggestion correctly, I think it might change the existing behavior or Azure.Core.  Is that your intention?

If that's not your intention, can you clarify your suggestion?  Perhaps you could share the specific lines of code you'd like to see changed?

Reply:
I don't think it effectively changes the behavior, just the implementation details. 

For instance, if `message.CancellationToken.IsCancellationRequested` is true I don't think we'll end up retrying no matter what is returned by `IsRetriable` as this would cause a failure in various other places along the way.  But, assuming we think it's important that the `ResponseClassifier` remains concerned with cancellation of the message, then it seems like a safe thing to check first as nothing else would make it retriable if cancellation has been requested.

Reply:
BTW, this is totally non-blocking - I'm just thinking through some of this logic as I see it come up in these reviews.

Reply:
Sounds good.  I'll go ahead and merge this, and let's follow up on the topic.  If I drop the ball, you should feel free to put up a PR with the changes you're proposing!  I know this tour through Azure.Core line by line is bringing up lots of questions about existing implementations!